gtars/src/uniwig/mod.rs

```diff
@@ -1,16 +1,722 @@
-//!
-//! # Uniwig BigWig generator
-
-//! The `uniwig` module is responsible for generating three BigWig output files based on a set of bed files. The generated files include a track for the start coordinates, a track for the end coordinates, and a track for the core coordinates.
-//!
-//! ## Under Construction
-//!
-//! This module is currently under construction. Stay tuned for more updates.
-//!
-//! ![Construction Sign](https://www.signoutfitters.com/images/products/detail/Workers_Symbol_Construction_Sign.png)
-//!
-//!
-
-pub fn run_uniwig() {
-    println!("Im running.")
+use clap::ArgMatches;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::path::Path;
+use std::fs::{File, OpenOptions};
+use std::error::Error;
+use clap::builder::OsStr;
+use flate2::read::GzDecoder;
+use ndarray::Array;
+use ndarray_npy::write_npy;
+
+
+pub mod cli;
+
+pub mod consts {
+    pub const UNIWIG_CMD: &str = "uniwig";
+
 }
+
+pub struct Chromosome {
+    chrom: String,
+    starts: Vec<i32>,
```

Comment:
Should this be `u32` since regions can't have negative start and ends?

Reply:
The starts/ends default to -1 if they cannot be parsed:
`let st = fields.next().and_then(|s| s.parse::<i32>().ok()).unwrap_or(-1);`

Reply:
I see... thats a bummer since its a lot memory allocated for a simple -1 :/

Reply:
On 64 bit architecture, isn't the usize larger than i32?

Reply:
Oh never mind, I misread the first comment.